homeassistant/components/homeassistant/triggers/time_pattern.py

```diff
@@ -75,7 +75,14 @@ async def async_attach_trigger(hass, config, action, automation_info):
     def time_automation_listener(now):
         """Listen for time changes and calls action."""
         hass.async_run_job(
-            action, {"trigger": {"platform": "time_pattern", "now": now}}
+            action,
+            {
+                "trigger": {
+                    "platform": "time_pattern",
+                    "now": now,
+                    "description": "time pattern",
```

Comment:
If they have multiple time patterns, would it be useful to show them which one matched so they don't have to do the math?

Reply:
I'm not sure how far to go with this. The amount of text shown could become a lot in some cases. E.g., would it make sense to show all the details for all the triggers, or the trigger index, or...??? I think to begin with it should be enough to narrow down to what type of trigger in the case of multiple types. And when there are multiple entities involved it's good to show which one's state change caused the trigger. For time based triggers, I'm not sure it's providing a lot to show more, but that's just my opinion. I'm good with starting with _something_ which is reasonable, and then possibly adding more later if people find they need the extra data.

Reply:
It is definitely hard to guess what users are going to have in their automation to know what they need to sort it out.  I ran though all of mine and didn't find any where I had multiple `time_pattern`s so it wouldn't help me.  The big win here is going to be which entity triggered it 👍